crates/orderbook/src/orderbook.rs

```diff
@@ -317,6 +324,16 @@ impl Orderbook {
         old_order: OrderUid,
         new_order: OrderCreation,
     ) -> Result<OrderUid, ReplaceOrderError> {
+        #[derive(Deserialize)]
```

Comment:
This function is getting a bit big. Maybe we could move the verification that it's actually okay to replace the order into a separate function.